src/config.c

```diff
@@ -33,12 +113,462 @@ const char *ChunkTypeToString(int options) {
+        if (!strcasecmp(encoding, UNCOMPRESSED_ARG_STR)) {
+            TSGlobalConfig.options &= ~SERIES_OPT_DEFAULT_COMPRESSION;
+            TSGlobalConfig.options |= SERIES_OPT_UNCOMPRESSED;
+        } else if (!strcasecmp(encoding, COMPRESSED_GORILLA_ARG_STR)) {
+            TSGlobalConfig.options &= ~SERIES_OPT_DEFAULT_COMPRESSION;
+            TSGlobalConfig.options |= SERIES_OPT_COMPRESSED_GORILLA;
+        } else {
+            *err = RedisModule_CreateStringPrintf(NULL, "Invalid encoding: %s", encoding);
+            return REDISMODULE_ERR;
+        }
+
+        return REDISMODULE_OK;
+    }
+
+    return REDISMODULE_ERR;
+}
+
+static long long getModernIntegerConfigValue(const char *name, void *privdata) {
+    if (!strcasecmp("ts-num-threads", name)) {
+        return TSGlobalConfig.numThreads;
+    } else if (!strcasecmp("ts-retention-policy", name)) {
+        return TSGlobalConfig.retentionPolicy;
+    } else if (!strcasecmp("ts-chunk-size-bytes", name)) {
+        return TSGlobalConfig.chunkSizeBytes;
+    } else if (!strcasecmp("ts-ignore-max-time-diff", name)) {
+        return TSGlobalConfig.ignoreMaxTimeDiff;
+    }
+
+    return 0;
+}
+
+static int setModernIntegerConfigValue(const char *name,
+                                       long long value,
+                                       void *data,
+                                       RedisModuleString **err) {
+    if (!strcasecmp("ts-num-threads", name)) {
+        TSGlobalConfig.numThreads = value;
+
+        return REDISMODULE_OK;
+    } else if (!strcasecmp("ts-retention-policy", name)) {
+        TSGlobalConfig.retentionPolicy = value;
+
+        return REDISMODULE_OK;
+    } else if (!strcasecmp("ts-chunk-size-bytes", name)) {
+        if (!ValidateChunkSize(NULL, value, err)) {
+            return REDISMODULE_ERR;
+        }
+
+        TSGlobalConfig.chunkSizeBytes = value;
+
+        return REDISMODULE_OK;
+    } else if (!strcasecmp("ts-ignore-max-time-diff", name)) {
+        if (value < 0) {
+            *err = RedisModule_CreateStringPrintf(
+                NULL, "Invalid value for `ts-ignore-max-time-diff`. Value must be non-negative");
+            return REDISMODULE_ERR;
+        }
+
+        TSGlobalConfig.ignoreMaxTimeDiff = value;
+
+        return REDISMODULE_OK;
+    }
+
+    return REDISMODULE_ERR;
+}
+
+bool RegisterModernConfigurationOptions(RedisModuleCtx *ctx) {
+    {
+        char *oldValue = CompactionRulesToString(TSGlobalConfig.compactionRules,
+                                                 TSGlobalConfig.compactionRulesCount);
+        if (!oldValue) {
+            oldValue = strdup("");
+        }
+
+        if (RedisModule_RegisterStringConfig(ctx,
+                                             "ts-compaction-policy",
+                                             oldValue,
+                                             REDISMODULE_CONFIG_DEFAULT |
+                                                 REDISMODULE_CONFIG_UNPREFIXED,
```

Comment:
I'm pretty sure that `DEFAULT` should not be combined with non-default flags. Regardless, this results in `UNPREFIXED` because `DEFAULT` is `0`